ipn/ipnlocal/local.go

```diff
@@ -1288,6 +1289,28 @@ func (b *LocalBackend) updateNetmapDeltaLocked(muts []netmap.NodeMutation) (hand
 // setExitNodeID updates prefs to reference an exit node by ID, rather
 // than by IP. It returns whether prefs was mutated.
 func setExitNodeID(prefs *ipn.Prefs, nm *netmap.NetworkMap) (prefsChanged bool) {
+	if exitNodeIDStr, _ := syspolicy.GetString(syspolicy.ExitNodeID, ""); exitNodeIDStr != "" {
+		exitNodeID := tailcfg.StableNodeID(exitNodeIDStr)
+		if prefs.ExitNodeID == exitNodeID && !prefs.ExitNodeIP.IsValid() {
+			return false
+		}
+		prefs.ExitNodeID = tailcfg.StableNodeID(exitNodeID)
+		prefs.ExitNodeIP = netip.Addr{}
+		return true
+	}
+
+	if exitNodeIPStr, _ := syspolicy.GetString(syspolicy.ExitNodeIP, ""); exitNodeIPStr != "" {
+		exitNodeIP, err := netip.ParseAddr(exitNodeIPStr)
+		if !exitNodeIP.IsValid() && err == nil {
+			if prefs.ExitNodeID == "" && prefs.ExitNodeIP == exitNodeIP {
+				return false
+			}
```

Comment:
This case shouldn't return, it needs to be converted into an `ExitNodeID` later in this func.